Amazon_Interview_Questions/Hard/Word_Ladder.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+
+	Problem link: https://leetcode.com/problems/word-ladder/
+	
+*/
+
+class Solution {
+public:
+    int ladderLength(string beginWord, string endWord, vector<string>& wordList) {  
+        unordered_set<string> myset;
+        
+        bool isPresent = false; //Checks if endWord is present in Dict
+        
+        for(auto word: wordList)
+        {
+            if(endWord.compare(word)==0)
+                isPresent = true;
+            myset.insert(word);    //Insert word in Dict
+        }
+        
+        if(isPresent==false)    //if endWord is not present
+            return 0;
+        
+        queue<string> q;
+        q.push(beginWord);
+        int depth = 0;
+        
+        while(!q.empty())
+        {
+            depth += 1;
+            int lsize = q.size();   //No. of elements at a level
+            while(lsize--)
+            {
+                string curr = q.front();
+                q.pop();
+                //check for all possible 1 depth words
+                for(int i=0;i<curr.length();++i)  //For each index
+                {
+                    string temp = curr;
+                    for(char c='a';c<='z';++c)  //Try all possible chars
+                    {
+                        temp[i] = c;
+                        if(curr.compare(temp)==0)
+                            continue;   //Skip the same word
+                        if(temp.compare(endWord)==0)
+                            return depth+1; //endWord found
+                        if(myset.find(temp)!=myset.end())
+                        {
+                            q.push(temp);
+                            myset.erase(temp);
+                        }
+                    }
+                }
+            }
+        }
+        return 0;
+    }
+};
```

Comment:
We already merged this in #3  no?
https://github.com/SaumyaSingh1/DSAImportantQuestionScratch/commit/17668089522d354a3d334acbe6b9beff3a279e10

Reply:
@SaumyaSingh1 we merged but that code is not from the last day.
Mam,
I am new to GitHub and its commands and open source contribution so might be an error from my side, I am sorry for that.
@SaumyaSingh1 Mam, one request, can you please see messages on LinkedIn, I have few queries, if you can clear them to me, I shall be obliged. Your 5 minutes of guidance every day, can shape my career.
Thank you.

Reply:
Sure thing!

Reply:
You can ask about the GitHub commands here as well :)